### ISSUE TITLE
should @computed('attr.something.@each.foo') provide attr.something?

### ISSUE THREAD

**Issue by stefanpenner** [rwjblue/ember-decorators#20]
example:

``` js
@computed('attr.model.@each.{something,somethingElse}')
foo(model) {
  // I think we want model === attr.model
  // not model == some useless proxy object
}
```

thoughts?

**Comment by rwjblue** [comment]
`@computed('attr.model.@each.{something,somethingElse}')` should pass the value of `attr.model` in to the callback function.

We should just pass in the portion prior to the `.@each` or `.[]`.

**Comment by stefanpenner** [comment]
> We should just pass in the portion prior to the .@each or .[].

my thoughts exactly +1

**Comment by stevesims** [comment]
I ran into this behaviour just yesterday, and frankly I was rather surprised to discover that I wasn't getting the value passed through to my computed function that I had specified as my dependent key.  The behaviour exhibited was non-obvious to me and unexpected.

Yes, I do realise that the behaviour from this issue is as has been documented, but I had skipped past that part of the documentation before.

My presumption was that by specifying it as the key I'd get passed `something.@each.foo`, i.e. an array of those `foo` values.  The first statement of my CP function ended up being a `get` for that key, and I ignored the `something` being passed into my function.

Had I wanted to work with `something`, but make sure that the CP is affected by the `foo` values changing then intuitively I'd have written my declaration to be `@computed('something', 'something.@each.foo') bar(something) {`.
